ext/TrixiMakieExt.jl

```diff
@@ -13,6 +20,387 @@ using Trixi: PlotData2DTriangulated, TrixiODESolution, PlotDataSeries, ScalarDat
+  trimesh = Vector{GeometryBasics.Mesh{3, Float32}}(undef, num_elements)
+  coordinates = zeros(Float32, num_plotting_nodes, 3)
+  for element in Base.OneTo(num_elements)
+    for i in Base.OneTo(num_plotting_nodes)
+      coordinates[i, 1] = x[i, element]
+      coordinates[i, 2] = y[i, element]
+      if set_z_coordinate_zero == false
+        coordinates[i, 3] = data.data[i, element]
+      end
+    end
+    trimesh[element] = GeometryBasics.normal_mesh(Makie.to_vertices(coordinates), makie_triangles)
+  end
+  plotting_mesh = merge([trimesh...]) # merge meshes on each element into one large mesh
+  return plotting_mesh
+end
+
+# Returns a list of `GeometryBasics.Point`s which can be used to plot the mesh, or a solution "wireframe"
+# (e.g., a plot of the mesh lines but with the z-coordinate equal to the value of the solution).
+function convert_PlotData2D_to_mesh_Points(pd::PlotData2DTriangulated{<:ScalarData};
+                                           set_z_coordinate_zero = false)
+  @unpack x_face, y_face, face_data = pd
+
+  if set_z_coordinate_zero
+    # plot 2d surface by setting z coordinate to zero.
+    # Uses `x_face` since `face_data` may be `::Nothing`, as it's not used for 2D plots.
+    sol_f = zeros(eltype(first(x_face)), size(x_face))
+  else
+    sol_f = face_data
+  end
+
+  # This line separates solution lines on each edge by NaNs to ensure that they are rendered
+  # separately. The coordinates `xf`, `yf` and the solution `sol_f`` are assumed to be a matrix
+  # whose columns correspond to different elements. We add NaN separators by appending a row of
+  # NaNs to this matrix. We also flatten (e.g., apply `vec` to) the result, as this speeds up
+  # plotting.
+  xyz_wireframe = GeometryBasics.Point.(map(x->vec(vcat(x, fill(NaN, 1, size(x, 2)))), (x_face, y_face, sol_f))...)
+
+  return xyz_wireframe
+end
+
+
+# We set the Makie default colormap to match Plots.jl, which uses `:inferno` by default.
+default_Makie_colormap() = :inferno
+
+# convenience struct for editing Makie plots after they're created.
+struct FigureAndAxes{Axes}
+  fig::Makie.Figure
+  axes::Axes
+end
+
+# for "quiet" return arguments to Makie.plot(::TrixiODESolution) and
+# Makie.plot(::PlotData2DTriangulated)
+Base.show(io::IO, fa::FigureAndAxes) = nothing
+
+# allows for returning fig, axes = Makie.plot(...)
+function Base.iterate(fa::FigureAndAxes, state=1)
+  if state == 1
+    return (fa.fig, 2)
+  elseif state == 2
+    return (fa.axes, 3)
+  else
+    return nothing
+  end
+end
+
+"""
+    iplot(u, mesh::UnstructuredMesh2D, equations, solver, cache;
+          plot_mesh=true, show_axis=false, colormap=default_Makie_colormap(),
+          variable_to_plot_in=1)
+
+Creates an interactive surface plot of the solution and mesh for an `UnstructuredMesh2D` type.
+
+Keywords:
+- variable_to_plot_in: variable to show by default
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+
```

Comment:
```suggestion
    This is an experimental feature and may change in future releases.
"""
function iplot end

```

From #1496

Reply:
https://github.com/trixi-framework/Trixi.jl/blob/c07a410c8e0603aaf769f7011d09e0d2f7c65771/ext/TrixiMakieExt.jl#L173

This is already there.

Reply:
Unless I am missing something 🤔?

Reply:
Yes, but the docstring is not connected to `iplot` - it just sits there in the middle of nowhere